File: game-server/src/handlers/login/player_login.rs
use crate::utils::time;
use sr_proto::{PlayerBasicInfo, PlayerLoginScRsp};

pub fn handle(_: &[u8]) -> PlayerLoginScRsp {
    PlayerLoginScRsp {
        basic_info: Some(PlayerBasicInfo {
            nickname: String::from("yulian"),
            level: 10,
            exp: 0,
            stamina: 240,
            mcoin: 1,
            hcoin: 1,
            scoin: 1,
            world_level: 1,
        }),
        server_timestamp_ms: time::cur_timestamp_ms(),
        retcode: 0,
        stamina: 240,
        ..Default::default()
    }
}

File: gameserver/handler/player/PlayerLogin.go
package player

import (
	"time"

	"github.com/yuvlian/qingque-sr/config/player"
	"github.com/yuvlian/qingque-sr/gameserver/session"
	"github.com/yuvlian/qingque-sr/pb"
	"github.com/yuvlian/qingque-sr/pb/cid"
)

func PlayerLogin(s *session.Session) error {
	rsp := &pb.PlayerLoginScRsp{
		BasicInfo: &pb.PlayerBasicInfo{
			Nickname:   player.Loaded.Username,
			Level:      61,
			Stamina:    240,
			WorldLevel: 5,
		},
		ServerTimestampMs: uint64(time.Now().UnixMilli()),
		Stamina:           240,
	}
	return s.Send(cid.PlayerLoginScRsp, rsp)
}
